kafka/src/main/java/io/micronaut/configuration/kafka/seek/KafkaSeekOperation.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.configuration.kafka.seek;
+
+import io.micronaut.core.annotation.NonNull;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.common.TopicPartition;
+
+import java.time.Duration;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Represents a {@code seek} operation that may be performed on a {@link Consumer}.
+ *
+ * <p>If successful, the operation will determine the next offset returned by
+ * {@link Consumer#poll(Duration)}.</p>
+ *
+ * @author Guillermo Calvo
+ * @see Consumer#seek(TopicPartition, long)
+ * @see Consumer#poll(Duration)
+ * @since 4.1
+ */
+public interface KafkaSeekOperation {
+
+    /**
+     * @return the topic name and partition number on which the seek should be performed.
+     */
+    @NonNull
+    TopicPartition topicPartition();
+
+    /**
+     * @return the offset type.
+     */
+    @NonNull
+    OffsetType offsetType();
+
+    /**
+     * @return the offset that should be used to perform the seek. Must be positive.
+     */
+    long offset();
+
+    /**
+     * @return the topic name on which the seek will be performed.
+     */
+    @NonNull
+    default String topic() {
+        return topicPartition().topic();
+    }
+
+    /**
+     * @return the partition number on which the seek will be performed.
+     */
+    default int partition() {
+        return topicPartition().partition();
+    }
+
+    /**
+     * Determines the interpretation of the {@link KafkaSeekOperation#offset()} value.
+     */
+    enum OffsetType {
+
+        /** The offset is absolute. */
+        ABSOLUTE,
+
+        /** The offset goes forward, relative to the current position. */
+        FORWARD,
+
+        /** The offset goes backward, relative to the current position. */
+        BACKWARD,
+
+        /** The offset goes forward, relative to the beginning of the partition. */
+        BEGINNING,
+
+        /** The offset goes backward, relative to the end of the partition. */
+        END,
+
+        /** The offset represents a Kafka timestamp. */
+        TIMESTAMP,
+    }
+
+    /**
+     * Convenience trait interface that can be used to create new {@link KafkaSeekOperation} instances.
+     */
+    interface Builder {
```

Comment:
Doesn't look like a builder to me, I would have those static methods in `KafkaSeekOperation`

Reply:
I agree the name could be improved, but I'm not a big fan of static methods. Besides, static methods make the code snippets more verbose than they need to be IMHO.

Anyway, I refactored the interface into static methods here: 944d29a02cf6d4aad3d9129121deafa12d647056

Reply:
WDYT @sdelamo

Reply:
I agree with @dstepanov I think getting rid of the builder is more consistent with API through the framework. 

For example: https://github.com/micronaut-projects/micronaut-security/blob/master/security/src/main/java/io/micronaut/security/authentication/AuthenticationResponse.java